packages/wrangler/src/init.ts

```diff
@@ -620,3 +688,191 @@ async function findPath(
+								},
+							],
+						};
+					}
+					break;
+				case "r2_bucket":
+					{
+						configObj["r2_buckets"] = [
+							...(configObj?.r2_buckets ?? []),
+							{ binding: binding.name, bucket_name: binding.bucket_name },
+						];
+					}
+					break;
+				case "service":
+					{
+						configObj["service"] = [
+							...(configObj?.service ?? []),
+							{
+								binding: binding.name,
+								service: binding.service,
+								environment: binding.environment,
+							},
+						];
+					}
+					break;
+				case "namespace":
+					{
+						configObj["namespaces"] = [
+							...(configObj?.namespaces ?? []),
+							{ binding: binding.name, namespace: binding.namespace },
+						];
+					}
+					break;
+				case "logfwdr":
+					{
+						configObj["logfwdr"] = {
+							// TODO: Messaging about adding schema file path
+							schema: "",
+							bindings: [
+								...(configObj?.logfwdr?.bindings ?? []),
+								{ name: binding.name, destination: binding.destination },
+							],
+						};
+					}
+					break;
+				case "wasm_module":
+					{
+						configObj["wasm_modules"] = {
+							...(configObj?.wasm_modules ?? {}),
+							[binding.name]: binding.part,
+						};
+					}
+					break;
+				case "text_blob":
+					{
+						configObj["text_blobs"] = {
+							...(configObj?.text_blobs ?? {}),
+							[binding.name]: binding.part,
+						};
+					}
+					break;
+				case "data_blob":
+					{
+						configObj["data_blobs"] = {
+							...(configObj?.data_blobs ?? {}),
+							[binding.name]: binding.part,
+						};
+					}
+					break;
+				default: {
+					// If we don't know what the type is, its an unsafe binding
+					// eslint-disable-next-line @typescript-eslint/no-explicit-any
+					if (!(binding as any)?.type) break;
+					configObj["unsafe"] = {
+						bindings: [...(configObj?.unsafe?.bindings ?? []), binding],
+					};
+				}
+			}
+
+			return configObj;
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		}, {} as Record<string, any>);
+
+	const durableObjectClassNames = bindings
+		.filter((binding) => binding.type === "durable_object_namespace")
+		.map(
+			(durableObject) =>
+				// We know that this is a Durable Object because of the filter above
+				(durableObject as Record<string, string>).class_name ?? ""
+		);
+
+	return {
+		compatibility_date: serviceEnvMetadata.script.compatibility_date,
+		routes,
+		usage_model: serviceEnvMetadata.script.usage_model,
+		migrations: {
+			tag: serviceEnvMetadata.script.migration_tag,
+			new_classes: durableObjectClassNames,
+		},
+		triggers: {
+			cron: cronTriggers.schedules.map((scheduled) => scheduled.cron),
+		},
+		env: environments
+			.filter((env) => env.environment !== "production")
+			// `env` can have multiple Enviroments, with different configs.
+			.reduce((envObj, { environment }) => {
+				return { ...envObj, [environment]: {} };
+				// eslint-disable-next-line @typescript-eslint/no-explicit-any
+			}, {} as Record<string, any>),
```

Comment:
Can this be better typed?

Reply:
Environment can be any permutation of the non-inheritable config